src/Lecture2.hs

```diff
@@ -48,7 +51,13 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct [] = 1
+lazyProduct list = go 1 list
```

Comment:
It looks to me that the first case is redundant because your helper function `go` will return `1` immediately if the given list is empty.

After you remove this case, you can also eta-reduce the top-level function ✂️ 
```suggestion
lazyProduct = go 1
```